modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx

```diff
@@ -0,0 +1,212 @@
+	namespace: string;
+}
+
+const AddStyleBookModalContent = ({
+	addStyleBookEntryURL,
+	closeModal,
+	frontendTokenDefinitionProviders = [],
+	namespace,
+}: AddStyleBookModalProps) =>{
+
+	const [active, setActive] = useState<boolean>(false);
+	const [loading, setLoading] = useState<boolean>(false);
+
+	const [errorMessage, setErrorMessage] = useState<string>();
+
+	const [name, setName] = useState<string>('');
+	const [themeId, setThemeId] =
+		useState<any>(frontendTokenDefinitionProviders[0]);
+
+	const formRef = useRef(null);
+
+	const handleFormError = (responseContent: any) => {
+		setErrorMessage(responseContent.error || '');
+	};
+
+	const handleSubmit = (event: any) => {
+		event.preventDefault();
+
+		const error = name.trim().length
+			? ''
+			: Liferay.Language.get('this-field-is-required');
+
+		if (error) {
+			setErrorMessage(error);
+
+			return;
+		}
+
+		setLoading(true);
+
+		const body = Liferay.Util.ns(namespace, {
+			name,
+			themeId: themeId.themeId,
+		});
+
+		fetch(addStyleBookEntryURL, {
+			body: objectToFormData(body),
+			method: 'POST',
+		})
+			.then((response) => response.json())
+			.then((responseContent) => {
+				if (responseContent.error) {
+					setLoading(false);
+					setErrorMessage(responseContent.error);
+				}
+				else if (responseContent.redirectURL) {
+					navigate(responseContent.redirectURL, {
+						beforeScreenFlip: closeModal,
+					});
+				}
+			})
+			.catch((response) =>
+				handleFormError(response)
+			);
+	};
+
+	const nameId = `${namespace}name`;
+	const themeIdId = `${namespace}tokenDefinition`;
+
+	return (
+		<>
+			<ClayModal.Header>
+				{Liferay.Language.get('add-style-book')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+
+				{errorMessage && (
+					<ClayAlert
+						displayType="danger"
+						onClose={() => {}}
+						title={Liferay.Language.get('error')}
+					>
+						{errorMessage}
+					</ClayAlert>
+				)}
+
+				<FieldBase
+					className="themeId"
+					helpMessage={Liferay.Language.get(
+						'the-style-book-will-be-created-based-on-the-selected-token-definition'
+					)}
+					id={themeIdId}
+					label={Liferay.Language.get('create-style-book-for')}
+				>
+					<Picker
+						active={active}
+						aria-labelledby="picker-label"
+						id={themeIdId}
+						onActiveChange={setActive}
+						onChange={(event: any) => {
+							const value = event.target.value;
+
+							setThemeId(frontendTokenDefinitionProviders[value]);
+						}}
+						onSelectionChange={setThemeId}
+						selectedKey={themeId.themeId}
+					>
+						{frontendTokenDefinitionProviders.map(
+							(frontendTokenDefinitionProvider) => (
+								<Option
+									key={frontendTokenDefinitionProvider.themeId}
+									textValue={frontendTokenDefinitionProvider.name}
+								>
+									{frontendTokenDefinitionProvider.name}
+								</Option>
+							)
+						)}
+					</Picker>
+				</FieldBase>
+
+				<ClayForm onSubmit={handleSubmit} ref={formRef}>
```

Comment:
Onsubmit isn't doing anything right now because the submit button isn't associated with this form. We can have the submit button submit the form even though it's not a child of the form by adding the form attribute to the button pointing to the id of the form.

So we should add an id on this form.

Reply:
Also, the form should wrap both inputs.